### DISCUSSION TITLE
Easier way to report "every road I traveled on was asphalt"

### DISCUSSION THREAD

**Discussion by endolith** [streetcomplete/StreetComplete#4439] [Ideas]
In my area, 99% of roads are asphalt. Clicking a bunch of times for each little segment is a lot of work and I'm not sure who even benefits from this.

 It's already recording my track. It would be nice if it just showed an overview of the whole trip and asked if any road surfaces are different, and if not, mark the whole thing at once.

**Comment by HolgerJeromin** [comment]
I would not be sure if really all the segment were the same. Easy to miss in the memory.

If you are not interested in this quest just deactivate it.

**Reply by endolith** [reply]
They are definitely all the same, I was just driving on them minutes ago.

**Reply by mnalis** [reply]
> [...] and I'm not sure who even benefits from this.

That is the valid point, @endolith. If you were ever routed to drive on `gravel` road when there was _just a little longer_ perfectly good `asphalt` road, you'd know the benefit immediately (especially if your transport vehicle is for example bicycle instead of all-terrain motor vehicle). But in advanced western world, the quest is often useful only outside of big cities (as roads in big cities in 1st-world countries are often - as you noticed - 99% asphalt or other similarly good pavement). 

So, if you only map in big city where 99% of the roads are asphalt, you should probably just disable that quest as suggested, and perhaps manually open a note (by long pressing on the map) to indicate those very rare exceptions (if you feel like it). 

Actually, you should always **disable _all_** quests that are bothersome to you, look spammy, or you do not see the value in them. That way, you'll have more fun and thus in the long run contribute much more than if the mapping starts feeling like drudgery because of such quests!

**Reply by endolith** [reply]
No, I want to fill out the information; I just don't want it to use such an inefficient interface.

**Comment by westnordost** [comment]
The app is not recording your track. It is just showing your path in the UI.

Anyway, regarding your question: No, this is not possible or planned to implement, but you can use an editor like iD to do that.

**Reply by endolith** [reply]
Yes, it is recording my track. That's how it displays my path in the UI. 😁

**Reply by westnordost** [reply]
🤷  yeah, but it is not saved anywhere. If you close and reopen the app, it's gone.

**Reply by simonpickering** [reply]
I also find this vaguely frustrating (and I am interested in the road surface as a cyclist), is there a webpage somewhere giving details of how to implement this using iD as suggested? i.e. I could quite happily chop up the GPX from a ride I've just done into sections that were definitely all asphalt/grass/gravel/etc and feed those in with the appropriate surface tag, leaving any edge or unremembered sections for further manual double-checking.

**Reply by FloEdelmann** [reply]
Since v44, you can now record GPX tracks and attach them to a note: https://github.com/streetcomplete/StreetComplete/releases/tag/v44.0

Long-press anywhere on the map and select "Start recording track".

**Reply by simonpickering** [reply]
Thanks, that's useful to know. While I realise it's not part of this project (it is related to the question though so might be useful), is there an established process (or some code) to (nearly-)automatically add the missing tags along a given GPX trace? (as I would plan to actually make the changes rather than just adding a note to be resolved) I know it's not quite as simple as it sounds as there can be ambiguity as to which "way" is being travelled, but presumably someone has looked at this and either decided it's too troublesome, or has come up with a bit of e.g. Python code? Even if it requires some manual interaction, it would presumably be faster than tapping on each icon in the app; but then again perhaps it's not faster/very useful in practice?

**Reply by mnalis** [reply]
If you use JOSM, you can load your GPX track, right click on it and choose `Download from OSM along this track`, choose sufficiently small radius to avoid false positives, set your filter to only show `highway=*` tags, and then select all highways with `ctrl-a` and remove all erroneous ones with `ctrl-click` on them (this might be time consuming part if you are not in area with few highways). When only correct highway parts are tagged, you can simply add `surface=asphalt` to all of them in `Tags` window. 
Only you can tell how useful it is in your case; I usually just start with nothing selected and manually ctrl-click every part of the highway I need to tag, because it is easier if there are just a dozen (or few dozen) parts.

**Reply by simonpickering** [reply]
Ah, great, thank you :)
